stdlib/public/Platform/musl.modulemap

```diff
@@ -2,110 +2,876 @@
 //
 // This source file is part of the Swift.org open source project
 //
-// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
 // Licensed under Apache License v2.0 with Runtime Library Exception
 //
 // See https://swift.org/LICENSE.txt for license information
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 //===----------------------------------------------------------------------===//
 
-module SwiftMusl [system] {
-  header "stdc-predef.h"
+module _musl [system] {
+  config_macros _GNU_SOURCE, _BSD_SOURCE, _XOPEN_SOURCE, _POSIX_SOURCE, _POSIX_C_SOURCE, _LARGEFILE64_SOURCE, _ALL_SOURCE, _DEFAULT_SOURCE, __STRICT_ANSI__
```

Comment:
That isn't a problem here, because bridging headers are an Xcode thing, and this isn't anything to do with Xcode. Also, the config macro declarations have actually shown up a number of build problems in Swift packages (which I've filed PRs for), so they're certainly useful.

I _think_ I'm right in saying that we _can_ use `config_macros` on Linux, *BSD, Musl and Windows. The place we can't is Darwin because of the bridging header issue.

Reply:
Yes. ObjC interop doesn't exist on non-Darwin, and even on Darwin if I were writing a program and not using Xcode, I doubt I'd use that flag — I'd probably make an explicit module map instead.